ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -398,12 +404,30 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       </div>;
     }
 
+    onResearchPurposeChange(checked: boolean) {
+      // If Checkbox is selected expand the research purpose categories
+      if (checked) {
+        this.setState({showResearchPurpose: true, selectResearchPurpose: true});
+        return;
+      }
+      this.setState({selectResearchPurpose: false});
```

Comment:
nit: I would prefer to use `else` here instead of using a return statement above. I thought this was a bug when I was reading through